File: PHP/MergeJoin/mergejoin.php
<?php
/**
 * MergeJoin algorithm implemented in PHP
 * @author: Sebastian Kim
 */
$row1 = fgetcsv($file1 = fopen('students.csv', r));
$row2 = fgetcsv($file2 = fopen('enrolment.csv', r));

while($row1 && $row2) {
	if($row1[0] < $row2[0])
		$row1 = fgetcsv($file1);
	else {
		if($row1[0] == $row2[0])
			echo "$row1[0], $row1[1], $row1[2], $row2[1], $row2[2]\n";
		$row2 = fgetcsv($file2);
	}	
}
?>

File: Python/MergeJoin/mergejoin.py
# MergeJoin algorithm implemented in Python 
# @author: Sebastian Kim

import csv
import sys

it1 = csv.reader(open('students.csv'))
it2 = csv.reader(open('enrolment.csv'))

row1 = it1.next()
row2 = it2.next()

while(row1 and row2):
	if(row1[0] < row2[0]):
		row1 = it1.next()
	else:
		if(row1[0] == row2[0]):
			print '{}, {}, {}, {}, {}'.format(row1[0],row1[1],row1[2],row2[1],row2[2])
		try:
			row2 = it2.next()
		except StopIteration:
			sys.exit(1);		
